src/main/java/net/openhft/chronicle/wire/LongConverter.java

```diff
@@ -67,6 +67,20 @@ static int maxParseLength(int based) {
      */
     long parse(CharSequence text);
 
+    /**
+     * Parses a part of the provided {@link CharSequence} and returns the parsed results as a
+     * {@code long} primitive.
+     *
+     * @param text character sequence containing the string representation of the value.
+     * @param beginIndex the beginning index, inclusive.
+     * @param endIndex the ending index, exclusive.
+     *
+     * @return the parsed {@code text} as an {@code long} primitive.
+     */
+    default long parse(CharSequence text, int beginIndex, int endIndex) {
+        return parse(text.toString().substring(beginIndex, endIndex));
```

Comment:
It seems that the original idea was to avoid extra object allocation (GC) but `text.toString()` is doing opposite.

Reply:
That's true, but this is just a default implementation, which is supposed to be overriden and then provide more efficient approach. Should probably be noted in javadoc.

Reply:
I've added a javadoc about it